source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -453,13 +453,21 @@ Establishing a Connection (Internal Implementation)
 
 Before a `Connection <#connection>`_ can be marked as either "available" or "in use", it
 must be established. This process involves performing the initial
-handshake, handling OP_COMPRESSED, and performing authentication.
+handshake, inspecting the value of ``maxWireVersion`` (to decide whether to
+use ``OP_MSG``(``maxWireVersion >= 6``) or ``OP_QUERY``(``maxWireVersion <6``)
+for future communications), handling OP_COMPRESSED, and performing
+authentication. It is necessary to inspect the value of ``maxWireVersion``
+because starting with MongoDB 6.0 using ``OP_QUERY`` for anything other than
+the handshake will result in an error. Once the driver has performed the
+handshake and inspected the value of ``maxWireVersion`` it MUST use
+``OP_MSG`` if supported.
```

Comment:
Should we link to the section in the OP_MSG spec instead of duplicating it? Alternatively, I would be more open to duplicating here once we make it more succinct (following my comment in OP_MSG). Or could we just remove this section altogether?

Reply:
I reduced it down significantly because the pseudocode offers some of the details